pkg/volume/gce_pd/gce_pd.go

```diff
@@ -76,25 +75,30 @@ func (plugin *gcePersistentDiskPlugin) NewMounter(spec *volume.Spec, pod *api.Po
 	return plugin.newMounterInternal(spec, pod.UID, &GCEDiskUtil{}, plugin.host.GetMounter())
 }
 
-func (plugin *gcePersistentDiskPlugin) newMounterInternal(spec *volume.Spec, podUID types.UID, manager pdManager, mounter mount.Interface) (volume.Mounter, error) {
-	// GCEPDs used directly in a pod have a ReadOnly flag set by the pod author.
-	// GCEPDs used as a PersistentVolume gets the ReadOnly flag indirectly through the persistent-claim volume used to mount the PV
+func getVolumeSource(spec *volume.Spec) (*api.GCEPersistentDiskVolumeSource, bool) {
 	var readOnly bool
+	var volumeSource *api.GCEPersistentDiskVolumeSource
 
-	var gce *api.GCEPersistentDiskVolumeSource
 	if spec.Volume != nil && spec.Volume.GCEPersistentDisk != nil {
-		gce = spec.Volume.GCEPersistentDisk
-		readOnly = gce.ReadOnly
+		volumeSource = spec.Volume.GCEPersistentDisk
+		readOnly = volumeSource.ReadOnly
 	} else {
-		gce = spec.PersistentVolume.Spec.GCEPersistentDisk
+		volumeSource = spec.PersistentVolume.Spec.GCEPersistentDisk
 		readOnly = spec.ReadOnly
```

Comment:
Ah right because the `PersistentVolumeClaimVolumeSource` defines a `ReadOnly` field that takes precedence. That's pretty confusing, we ignore the `ReadOnly` field on the PV object and let the `ReadOnly` field on the PVC object take precedence. Might be worth revisiting this logic in the future. Thanks.